composer/core/state.py

```diff
@@ -1017,29 +1085,36 @@ def load_optim_state(self, state_dict: Dict[str, Any]):
         """
         serialized_value = state_dict['optimizers']
         for optimizer in ensure_tuple(self.optimizers):
-            if type(optimizer).__qualname__ not in serialized_value:
+            if serialized_value is not None and type(optimizer).__qualname__ not in serialized_value:
                 warnings.warn(
                     f'{type(optimizer).__qualname__} is not in the state_dict. Its state will not be restored.',
                     category=UserWarning)
                 continue
-            optim_state_dict = serialized_value[type(optimizer).__qualname__]
+            optim_state_dict = serialized_value[type(optimizer).__qualname__] if serialized_value is not None else None
             if self.fsdp_enabled:
                 assert self.fsdp_state_dict_type is not None  # pyright
                 if version.parse(torch.__version__) < version.parse('1.13.0'):
                     raise RuntimeError('To use FSDP with Composer, you must use torch>=1.13.0.')
                 from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
                 log.debug(f'Loading FSDP optimizer with fsdp_state_dict_type={self.fsdp_state_dict_type}')
-                if not using_torch_2():
-                    optim_state_dict = _legacy_optim_state_dict_to_load(optim_state_dict=optim_state_dict,
-                                                                        model=self.model,
-                                                                        optim=optimizer,
-                                                                        state_dict_type=self.fsdp_state_dict_type)
+                # Loading FSDP monolith on rank 0 only requires FSDP.scatter_full_optim_state_dict
```

Comment:
@mvpatel2000 do you want to add a TODO here for when pytorch fixes?

Reply:
I have a JIRA tracking. I'll leave the todo out of the code